sync/src/synchronizer/block_fetcher.rs

```diff
@@ -40,41 +40,38 @@ impl<'a> BlockFetcher<'a> {
         self.synchronizer.peers().get_best_known_header(self.peer)
     }
 
-    pub fn last_common_header(&self, best: &HeaderView) -> Option<core::HeaderView> {
-        let tip_header = self.active_chain.tip_header();
-        let last_common_header = {
+    pub fn update_last_common_header(&self, best_known: &HeaderView) -> Option<core::HeaderView> {
+        // Bootstrap quickly by guessing a parent of our best tip is t
+        // Guessing wrong in either direction is not a problem.
+        let mut last_common =
             if let Some(header) = self.synchronizer.peers().get_last_common_header(self.peer) {
-                // may reorganized, then it can't be used
-                if header.number() > tip_header.number() {
-                    Some(tip_header)
-                } else {
-                    Some(header)
-                }
-            // Bootstrap quickly by guessing a parent of our best tip is the forking point.
-            // Guessing wrong in either direction is not a problem.
-            } else if best.number() < tip_header.number() {
-                let last_common_hash = self.active_chain.get_block_hash(best.number())?;
-                self.active_chain.get_block_header(&last_common_hash)
+                header
             } else {
-                Some(tip_header)
-            }
-        }?;
+                let tip_header = self.active_chain.tip_header();
+                let guess_number = min(tip_header.number(), best_known.number());
+                let guess_hash = self.active_chain.get_block_hash(guess_number)?;
+                self.active_chain.get_block_header(&guess_hash)?
+            };
 
         // If the peer reorganized, our previous last_common_header may not be an ancestor
         // of its current tip anymore. Go back enough to fix that.
-        let fixed_last_common_header = self
+        last_common = self
             .active_chain
-            .last_common_ancestor(&last_common_header, &best.inner())?;
+            .last_common_ancestor(&last_common, &best_known.inner())?;
 
-        Some(fixed_last_common_header)
+        self.synchronizer
+            .peers()
+            .set_last_common_header(self.peer, last_common.clone());
+
+        Some(last_common)
     }
 
     pub fn fetch(self) -> Option<Vec<Vec<packed::Byte32>>> {
-        trace!("[block downloader] BlockFetcher process");
+        let best_known = self.peer_best_known_header()?;
```

Comment:
The wrong call location of this function will cause the ibd multi-node download to fail, must after `try_update_best_known_with_unknown_header_list`

Reply:
Fixed: b32b91de1503264ad6fa5ca2cd82b82263c8d4a3